Ginger/GingerCoreNET/ActionsLib/APIModelActionConversion/ApiActionConversionUtils.cs

```diff
@@ -0,0 +1,461 @@
+                        }
+
+                        //Create WebAPIModel action
+                        ActWebAPIModel actApiModel = GetNewAPIModelAction(applicationModel.Guid, act);
+                        AddAppParameters(applicationModel, act, actApiModel, parameterizeRequestBody);
+
+                        activity.Acts.Insert(selectedActIndex + 1, actApiModel);
+                        actionIndex++;
+                        act.Active = false;
+                        if (!isModelExists)
+                        {
+                            WorkSpace.Instance.SolutionRepository.AddRepositoryItem(applicationModel);
+                        }
+                        else
+                        {
+                            WorkSpace.Instance.SolutionRepository.SaveRepositoryItem(applicationModel);
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Reporter.ToLog(eLogLevel.ERROR, "Error occurred while trying to convert action", ex);
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method will create the new api model action
+        /// </summary>
+        /// <param name="applicationModel"></param>
+        /// <param name="act"></param>
+        /// <returns></returns>
+        private ActWebAPIModel GetNewAPIModelAction(Guid applicationModelGuid, Act act)
+        {
+            AutoMapper.MapperConfiguration mapConfigUIElement = new AutoMapper.MapperConfiguration(cfg => { cfg.CreateMap<Act, ActWebAPIModel>(); });
+            ActWebAPIModel newActModel = mapConfigUIElement.CreateMapper().Map<Act, ActWebAPIModel>(act);
+            newActModel.APImodelGUID = applicationModelGuid;
+            newActModel.Active = true;
+            return newActModel;
+        }
+
+        /// <summary>
+        /// This method is used to check if the app parameter is already added in the list if not then it will add the parameter
+        /// </summary>
+        /// <param name="aPIModel"></param>
+        /// <param name="act"></param>
+        /// <param name="actApiModel"></param>
+        private void AddAppParameters(ApplicationAPIModel aPIModel, Act act, ActWebAPIModel actApiModel, bool parameterizeRequestBody)
+        {
+            try
+            {
+                ObservableList<AppModelParameter> lstParameters = null;
+                ObservableList<ActInputValue> actInputs = ((ActWebAPIBase)act).DynamicElements;
+                if (actInputs == null || actInputs.Count == 0)
+                {
+                    string requestBody = string.Empty;
+                    if (aPIModel.RequestBodyType == ApplicationAPIUtils.eRequestBodyType.TemplateFile)
+                    {
+                        string fileUri = aPIModel.TemplateFileNameFileBrowser.Replace("~\\", WorkSpace.Instance.Solution.ContainingFolderFullPath);
```

Comment:
Workspace.SolutionReposiory already has function which doing ~ replace, please use it